service/matching/matchingEngine.go

```diff
@@ -1193,19 +1201,15 @@ func (e *matchingEngineImpl) redirectToVersionedQueueForAdd(
 	if err != nil {
 		return nil, err
 	}
-	if userData.GetData().GetVersioningData() == nil {
-		if stamp == nil {
-			// queue is not versioned and neither is workflow, all good
-			return taskQueue, nil
-		}
-		return nil, errVersionedTaskForUnversionedQueue
-	}
+	data := userData.GetData().GetVersioningData()
 
-	// TODO: Here we have to distinguish between a new workflow (first wft), which we should
-	// assign to the default version), and a later wft, which we should leave on the
-	// unversioned queue. Do that in a follow-up PR.
+	if stamp == nil && data == nil {
+		// Unversioned workflow that started before this queue was versioned. Leave on
```

Comment:
This queue isn't versioned is data is nil, am I reading this comment wrong?

Reply:
as you pointed out, there's not really a concept of an unversioned queue anymore. but yeah, the comment is wrong...